apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaProducerInterceptor.java

```diff
@@ -59,8 +59,12 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
             next = next.next();
             record.headers().add(next.getHeadKey(), next.getHeadValue().getBytes());
         }
-        EnhancedInstance callbackInstance = (EnhancedInstance) allArguments[1];
-        if (null != callbackInstance) {
+        //fix bug: https://github.com/apache/skywalking/issues/4825
+        //when use lambda expression, not to generate inner class,
+        //    and not to trigger kafka CallBack class define, so allArguments[1] can't to cast EnhancedInstance
+        Object shouldCallbackInstance = allArguments[1];
+        if (null != shouldCallbackInstance && shouldCallbackInstance instanceof EnhancedInstance) {
```

Comment:
If you skip this, the cast exception is gone, yes. But, the callback will not be traced, right? I think you should create a new function to delegate the existing parameter, which could propagate the context and continue the trace in the callback invocation. WDYT?

Reply:
thanks for your advice.